arborcat.module

```diff
@@ -1,120 +1,216 @@
-        curl_setopt($ch, CURLOPT_URL, \Drupal::config('arborcat.settings')->get('pts_lockers'));
-        curl_setopt($ch, CURLOPT_HEADER, 0);
-        curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
-        curl_setopt($ch, CURLOPT_USERPWD, \Drupal::config('arborcat.settings')->get('lockers_pass'));
-        $result = curl_exec($ch);
-
-        $simpleXML = simplexml_load_string($result);
-        $i=1;
-        $s = $simpleXML->doors->children();
-        if($s->assign1==0&$s->assign2==0&$s->assign3==0&$s->assign4==0){
-           $available=false;
-        }
-        else if($s->assign1!==0||$s->assign2!==0||$s->assign3!==0||$s->assign4!==0){
-          $available=true;
-        }
+      if ($locker_server=='ptslockers') {
+          $ch = curl_init();
+          curl_setopt($ch, CURLOPT_URL, \Drupal::config('arborcat.settings')->get('pts_lockers'));
+          curl_setopt($ch, CURLOPT_HEADER, 0);
+          curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+          curl_setopt($ch, CURLOPT_USERPWD, \Drupal::config('arborcat.settings')->get('lockers_pass'));
+          $result = curl_exec($ch);
+
+          $simpleXML = simplexml_load_string($result);
+          $i=1;
+          $s = $simpleXML->doors->children();
+          if ($s->assign1==0&$s->assign2==0&$s->assign3==0&$s->assign4==0) {
+              $available=false;
+          } elseif ($s->assign1!==0||$s->assign2!==0||$s->assign3!==0||$s->assign4!==0) {
+              $available=true;
+          }
+      } elseif ($locker_server=='mcblockers') {
+          $ch = curl_init();
+          curl_setopt($ch, CURLOPT_URL, \Drupal::config('arborcat.settings')->get('mcb_lockers'));
+          curl_setopt($ch, CURLOPT_HEADER, 0);
+          curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+          curl_setopt($ch, CURLOPT_USERPWD, \Drupal::config('arborcat.settings')->get('lockers_pass'));
+          $result = curl_exec($ch);
+          if ($result != null) {
+              preg_match_all('%name="web_patroninserted_door.*?value="([\d]+?)"%s', $result, $matches);
+              if ($matches[1][0]>0) {
+                  $available = true;
+              }
+          }
       }
-        else if($locker_server=='mcblockers'){
-        $ch = curl_init();
-        curl_setopt($ch, CURLOPT_URL, \Drupal::config('arborcat.settings')->get('mcb_lockers'));
-        curl_setopt($ch, CURLOPT_HEADER, 0);
-        curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
-        curl_setopt($ch, CURLOPT_USERPWD, \Drupal::config('arborcat.settings')->get('lockers_pass'));
-        $result = curl_exec($ch);
-        preg_match_all('%name="web_patroninserted_door.*?value="([\d]+?)"%s', $result, $matches);
-        if ($matches[1][0]>0) {
-          $available = TRUE;
-        }
+      return $available;
+  }
+
+  /*
+   * returns a list of request pickup locations.
+   * If a branch location is passed in, that a filtered list of pickup locations is returned
+   */
+  function arborcat_pickupLocations($destLocation = null)
```

Comment:
I would change function name to underscore separated: arborcat_pickup_locations(), just to follow naming pattern